### ISSUE TITLE
Allow non-integer values for 0-1 type of axis, e.g., Italic

### ISSUE THREAD

**Issue by romashamin** [Mark2Mark/variable-font-preview#26]
![variable-font-preview-italic-axis@2x](https://github.com/Mark2Mark/variable-font-preview/assets/1148914/d7b07301-10c5-4f5a-86eb-2bf78eab3a93)

Seeing intermediate states of vector interpolation lets designers check for any path quirks. Currently, the slider for the Italic axis is discrete, allowing only the values 0 or 1, with no in-between values. This makes it impossible to check how the interpolation between upright and slanted shape works.

Please enable the slider to set non-integer values between 0 and 1 for such axes.

**Comment by Mark2Mark** [comment]
Thank you!

That has been done on purpose back then, usually that is how a UI would also implement it, if the axis has just the values 0 and 1. 

But I see your point, and also for example in web interfaces (CSS) the font can still be set to values between. 

May I ask, though: why do you define your italic space from 0 to 1, if you want to expose in-between values? Why not make the range bigger then?

**Comment by Mark2Mark** [comment]
Oh, yeah, didn't think about that. Thanks for the pointer. 
Would be great if you can find some clues, let me know 😊

- [ ] Consider differentiation between italic and slant axis
- [ ] Then deal with these according to this thread

**Comment by romashamin** [comment]
> Why not make the range bigger then?

According to [the spec](https://learn.microsoft.com/en-us/typography/opentype/spec/dvaraxistag_ital), `ital` is limited by 0 and 1:

> Values must be in the range 0 to 1. A value of 0 can be interpreted as “Roman” (non-italic); a value of 1 can be interpreted as (fully) italic.

So if the designer sets 12 instread of 1 as the second point that could lead to unpredictable results.

Also, Rainer Erich Scheichelbauer mentiones that [0-1 is still interpolatable](https://forum.glyphsapp.com/t/discrete-axis-support/28799/15) so it seems like the issue makes sense.

**Comment by florianpircher** [comment]
I believe the slnt axis is preferred if you want the user to pick in-between values. All axes are interpolatable, so even a range of 0–1 is technically interpolatable. In practice, such a range is understood by many apps to not interpolate.

**Comment by romashamin** [comment]
@florianpircher personally, I agree and would love to equip my typefaces with the `slnt` axis. It’s more flexible and accurately reflects that my typeface provides slanted letters that don’t change their shape as a true italic design implies.

Unfortunately, as far as I know, to date `ital` has better software support than `slnt`. So, I decided that using `ital`, especially if the typeface doesn’t include backslanted styles, would be a smart choice.

Regardless, choosing between `ital` and `slnt` for a particular typeface is a different question. Seeing interpolation between 0 and 1 in the plugin would help type designers like me test the interpolation for typefaces with an `ital` axis. Frankly, this feature was the main reason I bought the plugin.

**Comment by Mark2Mark** [comment]
Just for now, as long as the plugin does not support the fractions between 0 and 1, you can increase the axis range to something higher than 10 by just changing the max value of the max italic master(s). You might need to do that temporarily to check the interpolations in between and change it back after checking. 

I will need some time to implement that option and it will have to be a setting, because the current stepping for small ranges is also there for several reasons. So users will need to choose (and it increases the workload and maintenance, hence the waiting, sorry for that)

**Comment by Mark2Mark** [comment]
> Unfortunately, as far as I know, to date ital has better software support than slnt.

Have you opened a thread on the Glyphs forum about that? What are the limitations you experienced with the `slnt`? Maybe Georg has some insights of how those could be avoided instead of the use-ital-instead-hack. I mean the specs are there for some reasons I guess, so if the spec expects non-binary italics to be `slnt`, then it might be better to push the implementing applications to adhere to the spec. Just my 2 cents there. 🙃

**Comment by romashamin** [comment]
> Usually in design applications or alike, you won’t be able to set fractional values anyway.

Yes, but type designers don’t typically use design applications or alike to test their work-in-progress font vectors for quirks. Your plugin could be really helpful for that—that’s the idea.

If you’re not interested in implementing this, that’s fine. Just let me know, and I won’t bother you about it anymore.

**Comment by Mark2Mark** [comment]
> To match Glyphs’s width classes

You don’t have to put the master values to be the same as the width classes. 

That's why I am asking you those things, because it could be a setup that is not how it is supposed to be. And fixing quirks caused faulty setups in the plugin is not the right route to go. 

This has nothing to do with that I am not interested in fixing it changing things. It is about finding and fixing the actual cause. 

So for your particular problem with the Width axis: you can have your masters at for example 0, 200, 400 and still set the width classes to 2, 4, 5

(Those values are chosen randomly just as an example, but generally this is how you should set up your file)

This will also make your sliders way more granular without the need of fractions there. In fact, with my example you'd have a range of 400, which for a width axis is a lot, even if you have a super compressed master.

**Comment by Mark2Mark** [comment]
The axis location parameter is currently undergoing changes, due to the changes in Glyphs recently. There will be a switch between internal and external values once. 

For now it uses the same interpolation as Glyphs uses internally.

**Comment by romashamin** [comment]
I agree, I can switch to bigger values for the `width` axis, but the `ital` axis is still limited to a maximum value of 1.

**Comment by Mark2Mark** [comment]
Understood. Still have it on the list to see what I can do about the italic axis. Thanks for the reminder:)
